### ISSUE TITLE
fixed issue regarding referer url of compare page during adding product to wish list

### ISSUE THREAD

**Issue by ghost** [magento/magento2#18805]
### Preconditions (*)

1. Magento version 2.2.6
2. Magento version 2.3 beta
3. PHP v7.2
4. Wish list module with sample data

### Steps to reproduce (*)

1. Add product to compare.
2. Goto compare page and then Add product to Wishlist. (https://user-images.githubusercontent.com/32056137/46980523-734eb080-d0f2-11e8-929d-49706fca6441.png)
3. On successful addition of product to wishlist, the page is redirected to Wishlist home page, with a success message.
4. In that success Message, a link is provided to go back to the page from where the product is adding to wishlist Stating Click here to continue shopping. (https://user-images.githubusercontent.com/32056137/46980536-7f3a7280-d0f2-11e8-8a26-ec0f8ea7b641.png)
5. When you click this link it redirects to customer account page, rather going to compare page.(https://user-images.githubusercontent.com/32056137/46980544-83ff2680-d0f2-11e8-972e-8486041a8686.png)

### Expected result (*)

1. On clicking hyperlink on success message it should redirect to compare page.

### Actual result (*)

1. 
![customeraccount](https://user-images.githubusercontent.com/32056137/47495380-a111c300-d871-11e8-844f-b9ba6424525d.png)

2. It is redirecting to customer account page.
### 

Already fixed in PR: https://github.com/magento/magento2/pull/18637

**Comment by magento-engcom-team** [comment]
Hi @ayazwebkul. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@ayazwebkul do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by ghost** [comment]
@magento-engcom-team give me 2.2.6 instance

**Comment by magento-engcom-team** [comment]
Hi @ayazwebkul. Thank you for your request. I'm working on Magento 2.2.6 instance for you

**Comment by magento-engcom-team** [comment]
Hi @ayazwebkul, here is your [Magento instance](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18805-2-2-6/).
Admin access: [http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18805-2-2-6//admin](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18805-2-2-6//admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ghost** [comment]
yes, I was able to reproduce issue on vanilla Magento instance following steps to reproduce.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
#SQUASHTOBERFEST

**Comment by ghost** [comment]
fixed in PR: https://github.com/magento/magento2/pull/18637

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-95972`, `MAGETWO-95973` were created

**Comment by sudheernayak31** [comment]
@magento-engcom-team give me 2.3.0 instance

**Comment by magento-engcom-team** [comment]
Hi @sudheernayak31. Thank you for your request. I'm working on Magento 2.3.0 instance for you

**Comment by magento-engcom-team** [comment]
Hi @sudheernayak31, here is your [Magento instance](https://i-18805-2-3-0.instances.magento-community.engineering).
Admin access: [https://i-18805-2-3-0.instances.magento-community.engineering/admin](https://i-18805-2-3-0.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by engcom-Charlie** [comment]
Hello @ayazwebkul 

Thank you for contribution and collaboration!

We are not able to reproduce this issue on the lates `2.3-develop` branch by provided steps.

We are closing this issue due to _branch `2.2-develop` was closed and Magento no longer accepts pull requests and issues to the v2.2 release lines to focus all development efforts on v2.3._
 See [Accepted pull requests and ported code](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules) for more details

If you still faced this issue on `2.3` please create a new Issue with all required details according to [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines)
